src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -573,14 +595,37 @@ def __umount(self, pool, uuid):
         raise CallError(error) from None
 
     def __get_datasets(self, pool, uuid):
-        return [(f'{pool}/.system', '')] + [
-            (f'{pool}/.system/{i}', i) for i in [
-                'cores', 'samba4',
-                f'rrd-{uuid}', f'configs-{uuid}',
-                'webui', 'services',
-                f'netdata-{uuid}',
-            ]
-        ]
+        datasets = [(f'{pool}/.system', '', {})]
+        for i in [
+            'cores',
+            'samba4',
+            f'rrd-{uuid}',
+            f'configs-{uuid}',
+            'webui',
+            'services',
+            (
+                f'netdata-{uuid}', {
+                    'chown_config': {'uid': 999, 'gid': 997},
+                    'mountpoint': f'{SYSDATASET_PATH}/netdata',
+                    'props': {'canmount': 'noauto'},
+                    'create_paths': [
+                        {'path': '/var/log/netdata', 'uid': 999, 'gid': 997},
+                    ],
+                }
+            ),
```

Comment:
I think behavior is getting complex enough that it's a good idea to have a spec file (like fhs.py in our installer) that defines our system dataset contents and configuration.